src/SuperTokens.AspNetCore/HandshakeContainer.cs

```diff
@@ -85,28 +97,38 @@ public async Task<Handshake> GetHandshakeAsync(string? apiKey, string? cdiVersio
             }
         }
 
-        public async Task OnHandshakeChanged(string jwtSigningPublicKey, DateTimeOffset jwtSigningPublicKeyExpiration)
+        public async Task OnHandshakeChanged(IEnumerable<Net.SessionRecipe.KeyInfo>? jwtSigningPublicKeyList, string jwtSigningPublicKey, DateTimeOffset jwtSigningPublicKeyExpiration)
         {
             var handshake = _handshake;
             if (handshake == null)
             {
                 return;
             }
 
-            if (handshake.JwtSigningPublicKey.Equals(jwtSigningPublicKey, StringComparison.Ordinal) &&
-                handshake.JwtSigningPublicKeyExpiration.ToUnixTimeSeconds() == jwtSigningPublicKeyExpiration.ToUnixTimeSeconds())
+            var updatedSigningPublicKeyList = jwtSigningPublicKeyList != null
+                ? jwtSigningPublicKeyList.Select(keyInfo =>
+                    new AccessTokenSigningKey(keyInfo.publicKey,
+                        DateTimeOffset.FromUnixTimeMilliseconds(keyInfo.expirationTime),
+                        DateTimeOffset.FromUnixTimeMilliseconds(keyInfo.createdAt)
+                    ))
+                : new[] {
+                    new AccessTokenSigningKey(jwtSigningPublicKey,
+                        jwtSigningPublicKeyExpiration,
+                        DateTimeOffset.Now
+                )};
+
+            if (handshake.AccessTokenSigningPublicKeyList.Equals(updatedSigningPublicKeyList))
```

Comment:
Isn't this checking for reference equality of the underlying array? (Same question applies to a few lines down from here.)